core/src/main/java/software/amazon/awssdk/pagination/PaginatedItemsIterable.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.pagination;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+
+/**
+ * Iterable for the paginated items. This class can be used through iterate through
+ * all the items across multiple pages until there is no more response from the service.
+ *
+ * @param <ResponseT> The type of a single response page
+ * @param <ItemT> The type of paginated member in a response page
+ */
+@SdkProtectedApi
+public class PaginatedItemsIterable<ResponseT, ItemT> implements SdkIterable<ItemT> {
+
+    private final Paginated paginator;
```

Comment:
Should this have some kind of generics?

Reply:
Yep. I actually removed the Paginated interface as we are removing the main allItems() method from it. Not the response class directly extends SdkIterable instead of Paginated.
Ex: 
public final class ListTablesPaginator implements SdkIterable<ListTablesResponse>
instead of 
public final class ListTablesPaginator implements Paginated<ListTablesResponse, String>

Let me know if you have any concerns?

Reply:
+1